service/history/workflow_task_handler.go

```diff
@@ -211,6 +212,45 @@ func (handler *workflowTaskHandlerImpl) handleCommands(
 	return mutations, nil
 }
 
+func (handler *workflowTaskHandlerImpl) ensureUpdatesProcessed(
+	_ context.Context,
+	workflowTaskScheduledEventID int64,
+	workflowTaskHeartbeating bool,
+) error {
+
+	// If WT is already about to fail with another cause, don't check for unprocessed updates.
+	if handler.workflowTaskFailedCause != nil {
+		return nil
+	}
+
+	// If WT is a heartbeat WT, then it doesn't have to have messages.
+	if workflowTaskHeartbeating {
+		return nil
+	}
+
+	// If worker has just completed workflow with COMPLETE_WORKFLOW_EXECUTION command, then it doesn't have to have messages.
+	if !handler.mutableState.IsWorkflowExecutionRunning() {
+		return nil
+	}
+
+	unprocessedUpdateIDs := handler.updateRegistry.Unprocessed(workflowTaskScheduledEventID)
+	if len(unprocessedUpdateIDs) > 0 {
+		var updateIDs strings.Builder
+
+		for _, unprocessedUpdateID := range unprocessedUpdateIDs {
+			updateIDs.WriteString(unprocessedUpdateID)
+			updateIDs.WriteRune(',')
+		}
+
+		return handler.failWorkflowTask(
```

Comment:
If a user is running an old SDK and someone accidently sends an update request to that workflow their workflow is essentially stuck and can't  get unstuck until they update their SDK, are we OK with that? Seems kind of dangerous, especially if we add update to the UI.

Reply:
Thinking about this more, maybe the server should reject these updates with an error about how the worker didn't handle the update request and they may need to update their SDK?

Reply:
Yes, after few internal discussions we decided not to fail WT but reject updates.